src/Cli/dotnet/ParseResultExtensions.cs

```diff
@@ -64,15 +64,35 @@ public static string[] GetSubArguments(this string[] args)
             var runArgs = subargs.Contains("--") ? subargs.GetRange(subargs.IndexOf("--"), subargs.Count() - subargs.IndexOf("--")) : new List<string>();
             subargs = subargs.Contains("--") ? subargs.GetRange(0, subargs.IndexOf("--")) : subargs;
 
-            subargs.RemoveAll(arg => DiagOption.Aliases.Contains(arg));
-            if (subargs[0].Equals("dotnet"))
-            {
-                subargs.RemoveAt(0);
-            }
+            subargs = subargs.SkipWhile(arg => DiagOption.Aliases.Contains(arg) || arg.Equals("dotnet")).ToList();
```

Comment:
I believe it should only ever be in the first spot.

The reason I did it this way was to try to condense it into one 'skip' call. This is operating on arrays, and skipping just the opening pieces of the array can be a bit expensive if I do it too often. It shouldn't be a huge deal here just because the array should be short, but I don't see any harm in trying to make it a little more performant.

So what I'm really trying to say here is:
"If we start with dotnet, remove that. If we start with -d (or an alias for it), or if the second thing is -d (or an alias for it), skip that, too."

Do you think there's a better way to say that?

Reply:
OHHHH. I understand now. I hadn't used [SkipWhile](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.skipwhile?view=net-7.0) before, actually. I looked it up and understand now. 🤯 

> "If we start with dotnet, remove that. If we start with -d (or an alias for it), or if the second thing is -d (or an alias for it), skip that, too."
I completely misunderstood the exact problem being solved, but got that figured out after rereading the issue again. Based on your goal, `SkipWhile` definitely does your exactly this.

So, I took your solution and did a test. Instead of just changing this line, I simplified the entire method, because it looked really overblown:
```C#
public static string[] GetSubArguments2(this string[] args)
{
    var hasDashDash = false;
    // Bifurcates the arguments into two (potential) groups.
    // First group is before "--" and second group is after. If there is no "--", there's only a single group.
    var subCommandGroups = args.GroupBy(a => hasDashDash = hasDashDash || a.Equals("--")).ToArray();
    return subCommandGroups
        // Only modify elements before "--".
        .First()
        // Remove "dotnet" and diagnostic flags if they appear as the first or second element.
        .SkipWhile(arg => arg.Equals("dotnet") || DiagOption.Aliases.Contains(arg))
        // Remove the top-level command (ex: build, publish, run, etc.)
        .Skip(1)
        // Concatenates the subcommand elements (second group) if any exist.
        .Concat(hasDashDash ? subCommandGroups.Last() : []).ToArray();
}
```

Here's the test harness I made for it. It is being compared to the original code's method.
```C#
var command1 = "dotnet cake house --delicious --yum --d".Split(' ');
var command2 = "dotnet --d cheese --pantry -- --helpful --bacon --d".Split(' ');

Console.WriteLine("First:");
var answer1 = Something.GetSubArguments(command1).ToList();
answer1.ForEach(a => Console.Write($"{a} "));
Console.WriteLine();
Console.WriteLine("Second:");
var answer2 = Something.GetSubArguments(command2).ToList();
answer2.ForEach(a => Console.Write($"{a} "));
Console.WriteLine();

Console.WriteLine("First2:");
var answer12 = Something.GetSubArguments2(command1).ToList();
answer12.ForEach(a => Console.Write($"{a} "));
Console.WriteLine();
Console.WriteLine("Second2:");
var answer22 = Something.GetSubArguments2(command2).ToList();
answer22.ForEach(a => Console.Write($"{a} "));
Console.WriteLine();

Console.WriteLine($"First same: {answer1.SequenceEqual(answer12)}");
Console.WriteLine($"Second same: {answer2.SequenceEqual(answer22)}");
```

Here's the output. As you can see, the original removed the `--d` from `command1`, which was not desired. `command2` tests that `--d` after `dotnet` is being removed appropriately.
```
First:
house --delicious --yum
Second:
--pantry -- --helpful --bacon --d
First2:
house --delicious --yum --d
Second2:
--pantry -- --helpful --bacon --d
First same: False
Second same: True
```

Reply:
When you said the 'original', you meant the one before my change, right? Because I think I fixed the --d problem...

As far as your proposed version, I think it looks correct. I'm still concerned about setting a variable inside the GroupBy statement, but I don't really like the original version either...I'll push a proposed fix in a bit for you to look at that hopefully will make both of us happy.

I'm more in favor of the chained Skips and such. I'll include those in the change 🙂

Reply:
> When you said the 'original', you meant the one before my change, right?

Yeah, the code that is currently in the repo for this method.

> I'm still concerned about setting a variable inside the GroupBy statement

Using variable setting the way I am is more like a "hack", but it is the only simple way to allow for a flip-flop like that. Once it flips, it stays flipped. You need some kind of stored state for that to happen.

Personally, you'll see my style is to turn imperative logic into data-processing logic. Like, turning the code into SQL queries, basically. I find it simpler that way, but most don't seem to think that. I realized many years ago that most coding is just data manipulation/filtering so why not just use libraries/interfaces intended to process data? But I take it to the extreme sometimes. 😅

Reply:
Given `subCommandGroups` is Array, should use array indexer rather than first/last since group count is already 2. LINQ is pretty but not cheap.